app/src/main/java/com/example/calculator/MainActivity.kt

```diff
@@ -2,179 +2,84 @@ package com.example.calculator
+    private var strNumber = StringBuilder()
+    private lateinit var workingTextView: TextView
+    private lateinit var numberButtons: Array<Button>
+    private lateinit var operatorButtons: List<Button>
+    private var operator:Operator = Operator.NONE
+    private var isOperatorCliked: Boolean = false
+    private var operand1: Int = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-       }
-
-    fun numberAction(view: View)
-    {
-        if(view is Button)
-        {
-            if(view.text == ".")
-            {
-                if(canAddDecimal)
-                    workingTextView.append(view.text)
-
-                canAddDecimal = false
-            }
-            else
-                workingTextView.append(view.text)
-
-            canAddOperation = true
-        }
-    }
-
-    fun operationAction(view: View)
-    {
-        if(view is Button && canAddOperation)
-        {
-            workingTextView.append(view.text)
-            canAddOperation = false
-            canAddDecimal = true
-        }
-    }
-
-    fun allClear(view: View)
-    {
-        workingTextView.text = ""
-        resultsTextView.text = ""
-    }
-
-    fun delAction(view: View)
-    {
-        val length = workingTextView.length()
-        if(length > 0)
-            workingTextView.text = workingTextView.text.subSequence(0, length - 1)
-    }
-
-    fun equalsAction(view: View)
-    {
-        resultsTextView.text = calculateResults()
-        resultsTextView.text = calculateResults()
-    }
-
-    private fun calculateResults(): String
-    {
-        val digitsOperators = digitsOperators()
-
-        if(digitsOperators.isEmpty())
 
-            return ""
+        workingTextView = findViewById(R.id.workingTextView)
+        val workingTextView: TextView = findViewById(R.id.workingTextView)
 
-        val timesDivision = timesDivisionCalculate(digitsOperators)
-
-        if(timesDivision.isEmpty())
-
-            return ""
-
-        val result = addSubtractCalculate(timesDivision)
-
-        return result.toString()
-    }
-
-    private fun addSubtractCalculate(oldList: MutableList<Any>): Float
-    {
-        var result = oldList[0] as Float
-
-        for(i in oldList.indices)
-        {
-            if(oldList[i] is Char && i != oldList.lastIndex)
-            {
-                val operator = oldList[i]
-                val nextDigit = oldList[i + 1] as Float
-                if (operator == '+')
-                    result += nextDigit
-                if (operator == '-')
-                    result -= nextDigit
-            }
+        numberButtons = arrayOf(number9,number8,number7,number6, number5,number4,number3,number2,number1,number0)
+        for (i in numberButtons){
+            i.setOnClickListener {numberButtonClick(i)}
         }
-
-        return result
+        operatorButtons = listOf(DIV,MUL,SUB,ADD,CLEAR)
```

Comment:
operatorButtons is declared and used in one and same place
Does it need to exist as a field? Or at all? 
Consider using List<T>.foreach()